services/powchain/src/main/java/tech/pegasys/teku/services/powchain/Eth1ChainIdValidator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.services.powchain;
+
+import static tech.pegasys.teku.infrastructure.logging.StatusLogger.STATUS_LOG;
+
+import java.math.BigInteger;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.pow.Eth1Provider;
+import tech.pegasys.teku.util.config.Constants;
+
+public class Eth1ChainIdValidator {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Eth1Provider eth1Provider;
+
+  private final AsyncRunner asyncRunner;
+
+  public Eth1ChainIdValidator(Eth1Provider eth1Provider, AsyncRunner asyncRunner) {
+    this.eth1Provider = eth1Provider;
+    this.asyncRunner = asyncRunner;
+  }
+
+  public void validate() {
+    eth1Provider
+        .getChainId()
+        .thenAccept(this::onChainIdDetermined)
+        .exceptionallyCompose(
+            error -> {
+              LOG.error("Failed to get Eth1 chain id. Will retry.", error);
+              return asyncRunner.runAfterDelay(this::validate, 1, TimeUnit.SECONDS);
+            })
+        .always(() -> {});
+  }
+
+  private void onChainIdDetermined(BigInteger chainId) {
+    if (chainId.intValueExact() == Constants.DEPOSIT_CHAIN_ID) {
```

Comment:
This should be `!=` so we print the error if the chain ID does not match.
```suggestion
    if (chainId.intValueExact() != Constants.DEPOSIT_CHAIN_ID) {
```